limitador/src/storage/in_memory.rs

```diff
@@ -31,22 +31,40 @@ impl CounterStorage for InMemoryStorage {
 
     fn check_and_update(
         &self,
-        counters: HashSet<Counter>,
+        counters: &mut Vec<Counter>,
         delta: i64,
+        load_counters: bool,
     ) -> Result<Authorization, StorageErr> {
         // This makes the operator of check + update atomic
         let mut stored_counters = self.counters.write().unwrap();
 
-        for counter in counters.iter() {
-            if !Self::counter_is_within_limits(counter, stored_counters.get(counter), delta) {
-                return Ok(Authorization::Limited(
-                    counter.limit().name().map(|n| n.to_owned()),
-                ));
+        if load_counters {
+            let mut first_limited = None;
+            for counter in counters.iter_mut() {
+                let remaining =
+                    *stored_counters.get(counter).unwrap_or(&counter.max_value()) - delta;
+                counter.set_remaining(remaining);
```

Comment:
We're not updating the `expires_in` duration on this store... Not that I know how we'd do that. `TtlCache` doesn't seem to give us access to `InternalEntry.expiration` 🤔 
But I think we should get rid of it anyways... see #71

Reply:
what is the 'it' we should get rid of?

Reply:
`TtlCache`, but that's not for this PR...